### PR TITLE
[LPT] Support ONNX quantized models coming from ORT PTQ

### PR DESCRIPTION
**Pull request by eshoguli**
### Details:
 - *Support ONNX quantized models coming from ORT PTQ*

### Tickets:
 - *98890*

### CODE REVIEW

File: src/common/low_precision_transformations/tests/fuse_fake_quantize_transformation.cpp [hunk 1]
```diff
@@ -61,360 +51,88 @@ class FuseFakeQuantizeTransformation : public LayerTransformation,
...
         transformer
             .add<ngraph::pass::low_precision::FakeQuantizeDecompositionTransformation, ngraph::opset1::FakeQuantize>(
                 testValues.params);
-        transformer.transform(actualFunction);
-
+        transformer.add<ngraph::pass::low_precision::MaxPoolTransformation, ngraph::opset1::MaxPool>(testValues.params);
         transformer.add<ngraph::pass::low_precision::FakeQuantizeTransformation, ngraph::opset1::FakeQuantize>(
             testValues.params);
+        transformer.add<ngraph::pass::low_precision::FuseFakeQuantizeTransformation, ngraph::opset1::FakeQuantize>(
+            testValues.params);
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/common/low_precision_transformations/tests/fuse_fake_quantize_transformation.cpp:68]: I prefer to keep as is to reproduce transformation stack in functional tests. For example: to catch possible errors with float point calculations during FQ decomposition & fusing.

File: src/common/low_precision_transformations/tests/fuse_fake_quantize_transformation.cpp [hunk 2]
```diff
@@ -26,27 +21,22 @@ using namespace testing;
 using namespace ngraph;
 using namespace ngraph::pass;
 
+// clang-format off
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/common/low_precision_transformations/tests/fuse_fake_quantize_transformation.cpp:24]: removed

File: src/common/low_precision_transformations/tests/fuse_dequantize_to_fake_quantize_transformation.cpp
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/common/low_precision_transformations/tests/fuse_dequantize_to_fake_quantize_transformation.cpp:1]: Existing transformation was renamed to more appropriate name

File: src/common/low_precision_transformations/include/low_precision/fuse_fake_quantize.hpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * @ingroup ie_transformation_common_api
+ * @brief FuseFakeQuantizeTransformation fuse FakeQuantize operations.
+ *
+ * For more details about the transformation, refer to
+ * [FuseFakeQuantizeTransformation](@ref openvino_docs_OV_UG_lpt_FuseFakeQuantizeTransformation) page
+ * in the Inference Engine Developer Guide.
+ */
+class LP_TRANSFORMATIONS_API FuseFakeQuantizeTransformation : public LayerTransformation {
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/common/low_precision_transformations/include/low_precision/fuse_fake_quantize.hpp:23]: done

File: src/common/low_precision_transformations/src/fuse_fake_quantize.cpp [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include "itt.hpp"
+
+namespace ngraph {
+namespace pass {
+namespace low_precision {
+
+FuseFakeQuantizeTransformation::FuseFakeQuantizeTransformation(const Params& params) : LayerTransformation(params) {
+    MATCHER_SCOPE(FuseMultiplyToFakeQuantizeTransformation);
+    auto matcher = pattern::wrap_type<opset1::FakeQuantize>();
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/common/low_precision_transformations/src/fuse_fake_quantize.cpp:18]: done

File: src/common/low_precision_transformations/src/fuse_fake_quantize.cpp [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    auto fakeQuantize = ov::as_type_ptr<opset1::FakeQuantize>(root);
+
+    auto parent_output = fakeQuantize->get_input_source_output(0);
+    parent_output.remove_target_input(fakeQuantize->input(0));
+
+    for (auto& input : fakeQuantize->output(0).get_target_inputs()) {
+        input.replace_source_output(parent_output);
+    }
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/common/low_precision_transformations/src/fuse_fake_quantize.cpp:45]: done, thanks

File: src/common/low_precision_transformations/src/fuse_fake_quantize.cpp [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    this->register_matcher(m, callback);
+}
+
+bool FuseFakeQuantizeTransformation::transform(TransformationContext& context, ngraph::pattern::Matcher& m) {
+    const auto root = m.get_match_root();
+    if (!canBeTransformed(context, root)) {
+        return false;
+    }
+
+    auto fakeQuantize = ov::as_type_ptr<opset1::FakeQuantize>(root);
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/common/low_precision_transformations/src/fuse_fake_quantize.cpp:38]: correct, done

File: src/tests/functional/plugin/shared/src/low_precision_transformations/fuse_fake_quantize_transformation.cpp [hunk 7]
```diff
@@ -20,37 +20,80 @@ std::string FuseFakeQuantizeTransformation::getTestCaseName(const testing::TestP
...
+    auto split = [](const std::string& s, char delim) -> std::vector<std::string> {
+        std::vector<std::string> result;
+        std::stringstream ss(s);
+        std::string item;
+
+        while (std::getline(ss, item, delim)) {
+            result.push_back(item);
+        }
+        return result;
+    };
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/tests/functional/plugin/shared/src/low_precision_transformations/fuse_fake_quantize_transformation.cpp:68]: done, thanks

File: src/tests/functional/plugin/shared/src/low_precision_transformations/fuse_fake_quantize_transformation.cpp [hunk 8]
```diff
@@ -20,37 +20,80 @@ std::string FuseFakeQuantizeTransformation::getTestCaseName(const testing::TestP
...
+    auto absent = testValues.expected.absent;
+    auto int8_convolutions = 0ull;
+    for (auto it : rtInfo) {
+        const auto& nameIt = it.second.find("originalLayersNames");
+        const auto fused_name = nameIt->second.as<std::string>();
+
+        const auto names = split(fused_name, ',');
+        for (const auto& name : names) {
+            ASSERT_TRUE(absent.find(name) == absent.end());
+            exist.erase(name);
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/tests/functional/plugin/shared/src/low_precision_transformations/fuse_fake_quantize_transformation.cpp:80]: not neccessary, but why not, additional check was added

File: src/common/low_precision_transformations/src/fuse_fake_quantize.cpp [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+}
+} // namespace
+
+bool FuseFakeQuantizeTransformation::canBeTransformed(const TransformationContext& context, std::shared_ptr<Node> operation) const {
+    auto fakeQuantize = ov::as_type_ptr<opset1::FakeQuantize>(operation);
+    if ((fakeQuantize == nullptr) ||
+        (!ov::is_type<opset1::Constant>(fakeQuantize->get_input_node_ptr(1))) ||
+        (!ov::is_type<opset1::Constant>(fakeQuantize->get_input_node_ptr(2))) ||
+        (!ov::is_type<opset1::Constant>(fakeQuantize->get_input_node_ptr(3))) ||
+        (!ov::is_type<opset1::Constant>(fakeQuantize->get_input_node_ptr(4)))) {
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/common/low_precision_transformations/src/fuse_fake_quantize.cpp:79]: done, thanks

File: src/common/low_precision_transformations/src/fuse_fake_quantize.cpp [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+        return false;
+    }
+
+    const auto input_type = fakeQuantize->get_input_source_output(0).get_element_type();
+    const auto output_type = fakeQuantize->output(0).get_element_type();
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/common/low_precision_transformations/src/fuse_fake_quantize.cpp]: done

File: src/common/low_precision_transformations/src/fuse_fake_quantize.cpp [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+    for (const auto constant_value : constant_values) {
+        if (std::fabs(constant_value - value) > std::numeric_limits<float>::epsilon()) {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool check_intervals(const std::shared_ptr<opset1::FakeQuantize>& fakeQuantize) {
+    const auto element_type = fakeQuantize->output(0).get_element_type();
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/common/low_precision_transformations/src/fuse_fake_quantize.cpp]: done

File: src/common/low_precision_transformations/tests/fuse_fake_quantize_transformation.cpp [hunk 12]
```diff
@@ -61,360 +51,88 @@ class FuseFakeQuantizeTransformation : public LayerTransformation,
...
+               << testValues.expected.fakeQuantizeOnData1 << "_" << testValues.expected.fakeQuantizeOnData2 << "_";
         return result.str();
     }
 };
+// clang-format on
 
 TEST_P(FuseFakeQuantizeTransformation, CompareFunctions) {
     actualFunction->validate_nodes_and_infer_types();
-    auto res = compare_functions(actualFunction, referenceFunction, true, true);
+    auto res = compare_functions(actualFunction, referenceFunction, true, false);
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/common/low_precision_transformations/tests/fuse_fake_quantize_transformation.cpp]: good question, changed to `true`

File: src/tests/functional/plugin/shared/src/low_precision_transformations/fuse_fake_quantize_transformation.cpp [hunk 13]
```diff
@@ -20,37 +20,80 @@ std::string FuseFakeQuantizeTransformation::getTestCaseName(const testing::TestP
...
+    result << targetDevice << "_" <<
+        testValues.actual.precisionBefore << "_" <<
+        testValues.actual.fakeQuantizeOnData1 << "_" <<
+        testValues.actual.fakeQuantizeOnData2;
     return result.str();
 }
 
+InferenceEngine::Blob::Ptr FuseFakeQuantizeTransformation::GenerateInput(const InferenceEngine::InputInfo& info) const {
+    return FuncTestUtils::createAndFillBlob(info.getTensorDesc(), 20, 0, 1, 1);
+}
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply src/tests/functional/plugin/shared/src/low_precision_transformations/fuse_fake_quantize_transformation.cpp]: is not actual now, was removed

**Review by v-Golubev — APPROVED** [review]
LGTM

**Pull request merged**